Replace enzyme with react testing library

devinea opened the pull request:
Closes #833

Replaces all Enzyme-based tests in `@sap-ux/ui-components` with React Testing Library.

## Changes
- Migrated 44 test files from Enzyme to `@testing-library/react`
- Replaced `test-setup.cjs` (enzyme adapter config) with `test-setup.mjs` using RTL's `jest-dom` matchers
- Fixed a flaky `UITranslationInput` test (click-outside-callout) that was timing-dependent on CI

devinea committed:
wip

devinea committed:
wip

devinea committed:
packages/ui-components/test/unit/components/UIFlexibleTable.test.tsx

devinea committed:
wip

devinea committed:
packages/ui-components/test/unit/components/UIPersona.test.tsx

devinea committed:
UISearchBox.test.tsx

devinea committed:
UISection.test.tsx

devinea committed:
UISections.test.tsx

devinea committed:
packages/ui-components/test/unit/helper/MessageWrapper.test.tsx

devinea committed:
UIVirtualList.test.tsx

devinea committed:
UITreeDropdown.test.tsx

devinea committed:
UIToggleGroup.test.tsx

devinea committed:
UITooltip.test.tsx

devinea committed:
UITextfield.test.tsx

devinea committed:
lint fixes

devinea committed:
remove enzyme from renovate config

devinea committed:
Fix matchers - approach 1

devinea committed:
updates

devinea committed:
align used versions

devinea committed:
lock

devinea committed:
UICallout.test.tsx

devinea committed:
UIDropdown.test.tsx

devinea committed:
UIDialog.test.tsx

devinea committed:
UIContextualMenu.test.tsx

devinea committed:
UIChoiceGroup.test.tsx

devinea committed:
UIDatePicker.test.tsx

devinea committed:
UICombobox.test.tsx

devinea committed:
UIToggle.test.tsx

devinea committed:
UISections.test.tsx

devinea committed:
UITreeDropdown.test.tsx

devinea committed:
UIFlexibleTable.test.tsx

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme

* origin/main:
  chore: apply latest changesets
  Feat(preview-middleware): change file type of templates to ejs (#3441)
  chore: apply latest changesets
  feat(nodejs-utils): Add logging support to `CommandRunner` (#3452)
  chore: apply latest changesets
  feat: Add ADP manifest change wizard sub-generators (#3428)

devinea committed:
lint fixes

devinea committed:
UIFlexibleTable.test.tsx

devinea committed:
better coverage

devinea committed:
ULList test

devinea committed:
UIFlexibleTableActionButton.test.tsx

devinea committed:
add tests

devinea committed:
more tests

devinea committed:
improve coverage

devinea committed:
Add tests UITable-helper.test.tsx

devinea committed:
UIFlexibleTable.test.tsx

devinea committed:
lint fixes

devinea committed:
Fix tests based on review

devinea committed:
review feedback

devinea committed:
review feedback fixes

devinea committed:
refactor based on feedback

devinea committed:
more clean-up

devinea committed:
refactor change and update test

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme
* origin/main:
  chore: apply latest changesets
  chore/esbuild upgrade to 0.25.6 (#3456)

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme

* origin/main:
  chore: apply latest changesets
  Feat(ui5 info): add filter option for legacy free UI5 versions (#3455)

devinea committed:
Fix sonar issue

815are committed:
test: proposal with lost styles checks

proposal with lost styles checks

815are commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -189,221 +296,86 @@ describe('<UIDialog />', () => {
+                </UIDialog>
             );
+            const dialog = document.querySelector('.ms-Dialog-main');
+            expect(dialog).toBeInTheDocument();
```
my proposal would be https://github.com/SAP/open-ux-tools/pull/3459/commits/27fa52ff24a13693f6368b8ec503fd13669b5eb9

815are committed:
test: cleanup callout test

cleanup callout test

815are committed:
test: style check improvements

style check improvements

815are committed:
fix: lint

lint

815are committed:
fix: lint

lint

815are committed:
test: adjustments

adjustments

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme
* origin/main:
  chore(deps): update dependency @types/inquirer-autocomplete-prompt to v2.0.2 (#1651)
  chore: apply latest changesets
  Fix UI5 1.120.x linting for basic template, fix warnings (#3434)
  fix: remove filtering of parameters for inbounds (#3440)
  chore: apply latest changesets
  fix: provide templates path parameter to handle different runtime context (#3464)
  chore: apply latest changesets
  fix(fiori-gen): updates FioriAppGeneratorPromptSettings type (#3474)
  chore: apply latest changesets
  fiori-sub-gen: update `FioriGeneratorPromptExtension` types (#3444)
  fix(deps): update jest monorepo to v30 (major) (#3362)
  chore: apply latest changesets
  fix(repo-app-import-sub-generator): set only css: [] in manifest to fix lint error (#3468)
  chore: apply latest changesets
  feat(odata-service-inq): set analytical table when supported by main entity (#3467)
  Project modules update 22.07.2025 (4690) (#3462)
  chore: apply latest changesets
  move headless AppConfig type and associated types to fiori gen shared module (#3466)
  fix: sonar issue for prompting examples (#3460)
  chore: delete cards editor middlewares (#3454)

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme

* origin/main:
  chore: apply latest changesets
  update lock file
  chore(deps): update sonarsource/sonarqube-scan-action action to v5 (#3381)
  chore(deps): update dependency supertest to v7 (#3144)
  chore(deps): update babel monorepo (#3143)
  fix(deps): update dependency @ui5/manifest to v1.76.0 (#3471)
  chore: apply latest changesets
  Revert "fix(security): form-data upgrade via axios upgrades"
  fix(security): form-data upgrade via axios upgrades
  chore(deps): update dependency fast-check to v2.25.0 (#1316)
  chore(deps): update dependency puppeteer-core to v22.15.0 (#3152)
  fix(deps): update dependency @sap-ux/annotation-converter to v0.10.3 (#3270)
  chore: apply latest changesets
  tbi(axios-extension): Improves v4 services load times (#3478)

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme
* origin/main: (286 commits)
  chore: apply latest changesets
  feat(odata-service-inquirer): 35947 default table type (#3725)
  chore: apply latest changesets
  Fix/35934/incorrect entity list rich text editor (#3724)
  fix: Remove CF flow from the ADP Generator and fix template resolution problem (#3735)
  chore: apply latest changesets
  fix: vocabulary update (#3733)
  chore: apply latest changesets
  feat(fiori-mcp-server): add fiori_development_portal_extension.md, etc... (#3717)
  fix: schedule ui integration tests on every workday (#3620)
  chore: apply latest changesets
  feat(fpm-writer): table building block custom columns (#3665)
  chore: apply latest changesets
  fix: Project type appears twice under system and password prompts when cloud system is selected. (#3706)
  chore: apply latest changesets
  fix(adp)(vscode): SAP Fiori Launchpad Configuration step appears multiple times.  (#3683)
  chore: apply latest changesets
  update axios version (#3732)
  chore: apply latest changesets
  fix(fiori-annotation-api): wrong origin range (#3729)
  ...

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme
* origin/main: (85 commits)
  lint error handling for forks
  fix: update repository reference in checkout step of lint workflow (#3808)
  chore: apply latest changesets
  chore: add badge to changelog and deep link to package to readme.md files (#3791)
  chore: apply latest changesets
  chore: update pipeline to publish extensions as draft (#3807)
  chore: apply latest changesets
  fix(fiori-annotation-api): move and delete  (#3785)
  chore: apply latest changesets
  generalise pipeline config (#3804)
  chore: apply latest changesets
  test config updates (#3803)
  chore: apply latest changesets
  correct sap systems extension naming in config (#3802)
  chore: apply latest changesets
  pipeline config test for extension (#3801)
  chore: apply latest changesets
  fix(preview-middleware): i18n handling for cards generator (#3736)
  chore: apply latest changesets
  feat(preview-middleware): enable card generator for CAP projects (#3737)
  ...

devinea committed:
temp until https://github.com/SAP/open-ux-tools/pull/3771 is merged.

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme
* origin/main: (31 commits)
  chore: apply latest changesets
  fix: improve handling for saving systems (#3828)
  chore: apply latest changesets
  chore: Update `@sap/cf-tools` version to latest to avoid security issues (#3809)
  chore: apply latest changesets
  feat(fiori-docs-embeddings): add @sap-ux/create/README.md as resource (#3779)
  fix: run tests on windows (#3823)
  chore: apply latest changesets
  update building block svg (#3826)
  chore: apply latest changesets
  feat(create): auto generate readme on build (#3760)
  chore: apply latest changesets
  Resolve Transport Request Checks for Download Quick Fiori App Deploy (#3819)
  test: v4 project builder (#3682)
  chore: apply latest changesets
  fix: bump specVersion to v4.0 (#3744)
  chore: apply latest changesets
  test: Integration test for fiori MCP using promptfoo (#3705)
  chore: apply latest changesets
  fix: move message to store creds prompt, tests, changeset (#3820)
  ...

devinea committed:
merge from main

devinea committed:
fix tests

devinea committed:
compare styles in tests

devinea committed:
test: enhance UIDialog and UIToggle styles validation

devinea committed:
fixes

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme

* origin/main:
  chore: apply latest changesets
  tbi: generator updates to support yo v5 (#3950)
  chore: apply latest changesets
  feat(ui-component): add StarBig icon (#3961)

devinea committed:
cleanup

815are committed:
test: callout test

callout test

815are commented on packages/ui-components/test/unit/components/UICallout.test.tsx:
```diff
@@ -128,14 +178,86 @@ describe('<UICallout />', () => {
+                );
+
+                const dummyElement = container.querySelector('.dummy');
+                if (dummyElement) {
```
adjustment - 34332fda52b13a96248bf5708e909633f49daec5

815are committed:
fix: cleanup

cleanup

815are committed:
test: validate and simplify checkbox

validate and simplify checkbox

815are committed:
test: adjust test for contextual menu

adjust test for contextual menu

815are committed:
test: adjust datepicker test

adjust datepicker test

815are committed:
test: adjust tests for default button

adjust tests for default button

815are commented:
validating tests:
- [x] UIActionButton.test.tsx c52a493b2f215db7f241ec87524981bc4ec7effb
- [x] UICallout.test.tsx 34332fda52b13a96248bf5708e909633f49daec5
- [x] UICheckbox.test.tsx  dddd496f3e5ca6eea908b998d3ee1c528f1cb6ec 44b49f9e449002ad1f5a21ef4ed4d6673d927378
- [x] UIChoiceGroup.test.tsx d00da66 da7fcb856a3985e677e868ec3152765d483444fe
- [x] UICombobox.test.tsx 44b49f9e449002ad1f5a21ef4ed4d6673d927378 7ccc3d3ca217d2002bcbf20fd119f10b6d24d78c
- [x] UIContextualMenu.test.tsx ae042fe1096ab8318cafef5de244019830e2af1f 3c1ed3b4da39221426c528086a43c4c91a4e9f00
- [x] UIDatePicker.test.tsx 389db9a833554864d73501fcac4d53a38a919e6d
- [x] UIDefaultButton.test.tsx 0526f882630bed58310b61b49fd7b7f43758ad0e da7fcb856a3985e677e868ec3152765d483444fe
- [x] UIDialog.test.tsx f949e15acf61b88cb65407b9bee5068a6793b7a3 1504b0a55fa229df9ed53b004c66c94fb799e7ac 7ccc3d3ca217d2002bcbf20fd119f10b6d24d78c
- [x] UIDropdown.test.tsx 44b49f9e449002ad1f5a21ef4ed4d6673d927378 7ccc3d3ca217d2002bcbf20fd119f10b6d24d78c
- [x] UIFlexibleTable.test.tsx 995d72a6b55462402cc270a28a5500e63dff99c7
- [x] UIFlexibleTableActionButton.test.tsx
- [x] UIHighlightMenuOption.test.tsx
- [x] UIIcon.test.tsx 44b49f9e449002ad1f5a21ef4ed4d6673d927378
- [x] UILabel.test.tsx da7fcb856a3985e677e868ec3152765d483444fe
- [x] UILink.test.tsx 44b49f9e449002ad1f5a21ef4ed4d6673d927378 08b092573983fc69989023101bc66f0e577a2715
- [x] UIList.test.tsx
- [x] UILoader.test.tsx 264a39ea58d88f265ab41eea109b41ec0a5a1169
- [x] UIMessageBar.test.tsx 44b49f9e449002ad1f5a21ef4ed4d6673d927378 08b092573983fc69989023101bc66f0e577a2715
- [x] UIPersona.test.tsx 264a39ea58d88f265ab41eea109b41ec0a5a1169
- [x] UISearchBox.test.tsx 44b49f9e449002ad1f5a21ef4ed4d6673d927378 264a39ea58d88f265ab41eea109b41ec0a5a1169
- [x] UISection.test.tsx 44b49f9e449002ad1f5a21ef4ed4d6673d927378 986a660518d0030b10a5fa7a99cde0471420f7cc
- [x] UISections.test.tsx  44b49f9e449002ad1f5a21ef4ed4d6673d927378 a5b7f5c84153f09b4aa63e5859ab4bef5bfa4421
- [x] UISeparator.test.tsx
- [x] UISmallButton.test.tsx 08b092573983fc69989023101bc66f0e577a2715
- [x] UISplitButton.test.tsx 3c1ed3b4da39221426c528086a43c4c91a4e9f00
- [x] UISplitter.test.tsx 44b49f9e449002ad1f5a21ef4ed4d6673d927378 3c1ed3b4da39221426c528086a43c4c91a4e9f00
- [x] UITable-helper.test.tsx 7ccc3d3ca217d2002bcbf20fd119f10b6d24d78c
- [x] UITable.test.tsx fb858b580df52b8d919cc65810900184239db178
- [x] UITextfield.test.tsx da7fcb856a3985e677e868ec3152765d483444fe 44b49f9e449002ad1f5a21ef4ed4d6673d927378 7ccc3d3ca217d2002bcbf20fd119f10b6d24d78c
- [x] UIToggle.test.tsx da7fcb856a3985e677e868ec3152765d483444fe 44b49f9e449002ad1f5a21ef4ed4d6673d927378 3c1ed3b4da39221426c528086a43c4c91a4e9f00
- [x] UIToggleGroup.test.tsx
- [x] UITooltip.test.tsx 44b49f9e449002ad1f5a21ef4ed4d6673d927378 08b092573983fc69989023101bc66f0e577a2715 7ccc3d3ca217d2002bcbf20fd119f10b6d24d78c
- [x] UITreeDropdown.test.tsx 7ccc3d3ca217d2002bcbf20fd119f10b6d24d78c
- [x] UIVirtualList.test.tsx
- [x] MessageWrapper.test.tsx 3c1ed3b4da39221426c528086a43c4c91a4e9f00

815are committed:
test: adjust dialog tests 1

adjust dialog tests 1

815are committed:
test: adjust dialog tests 2

adjust dialog tests 2

815are committed:
fix: adjust style calculation

adjust style calculation

devinea committed:
Merge branch 'main' into replace_enzyme

devinea committed:
Merge origin/main into replace_enzyme

Resolved conflicts:
- packages/ui-components/package.json: kept newer versions from main, removed enzyme/enzyme-adapter deps
- packages/ui-prompting/package.json: kept newer storybook versions from main, removed @storybook/addons (deprecated)
- packages/preview-middleware-client/package.json: upgraded ui5-tooling-modules to 3.35.0
- renovate.json5: added new dependency groups from main (mem-fs-editor, inquirer, memfs, yeoman, react-markdown, express), dropped enzyme/jest-environment-jsdom constraints
- pnpm-lock.yaml: accepted main version
- test files (UIActionButton, UICallout, UICombobox, UIDefaultButton, UIDropdown, test-setup.js, snapshot): kept replace_enzyme RTL-based implementations

devinea committed:
fixes after merge from main

devinea committed:
fixes

devinea committed:
chore: update pnpm-lock.yaml after merge from main

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme

* origin/main: (210 commits)
  chore: apply latest changesets
  Fix: CDS multiple service targets in the same file, pointers bug fix (#4796)
  chore: apply latest changesets
  fix(odata-downloader): Fix hierarchy issue (previous PR missed commit) (#4818)
  chore: apply latest changesets
  fix(fiori-mcp-server): fix credential loading, Zowe keyring bundling, and metadata formatting (#4776)
  chore: apply latest changesets
  fix(ui5-test-writer): use js files for FPM opa tests (#4817)
  chore: apply latest changesets
  feat(ui5-test-writer): TypeScript OPA5 test generation for Fiori Elements V4 (#4680)
  chore: apply latest changesets
  fix(create): 38324 update system error message 2 (#4815)
  chore: apply latest changesets
  fix(odata-downloader): Multiple fixes and adds ouput tab link (`Show Logs`) to the odata downloader (#4810)
  chore: apply latest changesets
  fix(abap-deploy): write builder.resources.excludes to base ui5.yaml (#4770)
  chore: apply latest changesets
  feat: switch build from tsc to esbuild bundling (#4807)
  chore: apply latest changesets
  feat(ui-icons): add ValidationWarning icon to UiIcons (#4804)
  ...

# Conflicts:
#	.github/workflows/integration-tests.yml
#	.github/workflows/lint.yml
#	.github/workflows/sonar.yml
#	.github/workflows/thank-you-sync.yml
#	.vscode/launch.json
#	docs/tools-offline-instructions/info/projectModules.json
#	examples/fe-fpm-cli/CHANGELOG.md
#	examples/fe-fpm-cli/package.json
#	examples/odata-cli/CHANGELOG.md
#	examples/odata-cli/package.json
#	examples/simple-generator/CHANGELOG.md
#	examples/simple-generator/package.json
#	examples/ui-prompting-examples/CHANGELOG.md
#	examples/ui-prompting-examples/package.json
#	package.json
#	packages/abap-deploy-config-inquirer/CHANGELOG.md
#	packages/abap-deploy-config-inquirer/package.json
#	packages/abap-deploy-config-sub-generator/CHANGELOG.md
#	packages/abap-deploy-config-sub-generator/package.json
#	packages/abap-deploy-config-writer/CHANGELOG.md
#	packages/abap-deploy-config-writer/package.json
#	packages/abap-deploy-config-writer/test/unit/__snapshots__/index.test.ts.snap
#	packages/adp-flp-config-sub-generator/CHANGELOG.md
#	packages/adp-flp-config-sub-generator/package.json
#	packages/adp-tooling/CHANGELOG.md
#	packages/adp-tooling/package.json
#	packages/annotation-generator/CHANGELOG.md
#	packages/annotation-generator/package.json
#	packages/annotation-generator/tsconfig.json
#	packages/app-config-writer/CHANGELOG.md
#	packages/app-config-writer/package.json
#	packages/axios-extension/CHANGELOG.md
#	packages/axios-extension/package.json
#	packages/backend-proxy-middleware-cf/CHANGELOG.md
#	packages/backend-proxy-middleware-cf/package.json
#	packages/backend-proxy-middleware/CHANGELOG.md
#	packages/backend-proxy-middleware/package.json
#	packages/btp-utils/CHANGELOG.md
#	packages/btp-utils/package.json
#	packages/btp-utils/src/app-studio.ts
#	packages/cap-config-writer/CHANGELOG.md
#	packages/cap-config-writer/package.json
#	packages/cds-odata-annotation-converter/CHANGELOG.md
#	packages/cds-odata-annotation-converter/package.json
#	packages/cf-deploy-config-inquirer/CHANGELOG.md
#	packages/cf-deploy-config-inquirer/package.json
#	packages/cf-deploy-config-sub-generator/CHANGELOG.md
#	packages/cf-deploy-config-sub-generator/package.json
#	packages/cf-deploy-config-sub-generator/test/fixtures/sap-ux-test/router/package.json
#	packages/cf-deploy-config-writer/CHANGELOG.md
#	packages/cf-deploy-config-writer/package.json
#	packages/control-property-editor/tsconfig.eslint.json
#	packages/control-property-editor/tsconfig.json
#	packages/create/CHANGELOG.md
#	packages/create/package.json
#	packages/create/src/cli/add/system.ts
#	packages/deploy-config-generator-shared/CHANGELOG.md
#	packages/deploy-config-generator-shared/package.json
#	packages/deploy-config-sub-generator/CHANGELOG.md
#	packages/deploy-config-sub-generator/package.json
#	packages/deploy-tooling/CHANGELOG.md
#	packages/deploy-tooling/package.json
#	packages/deploy-tooling/test/__mocks__/index.ts
#	packages/environment-check/CHANGELOG.md
#	packages/environment-check/package.json
#	packages/eslint-plugin-fiori-tools/CHANGELOG.md
#	packages/eslint-plugin-fiori-tools/package.json
#	packages/eslint-plugin-fiori-tools/src/index.ts
#	packages/fe-fpm-writer/CHANGELOG.md
#	packages/fe-fpm-writer/package.json
#	packages/fe-fpm-writer/tsconfig.json
#	packages/fiori-annotation-api/CHANGELOG.md
#	packages/fiori-annotation-api/package.json
#	packages/fiori-app-sub-generator/CHANGELOG.md
#	packages/fiori-app-sub-generator/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v2/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v2/ui5-local.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v2/ui5-mock.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v2/ui5.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v2_no_opts/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v2_no_opts/ui5-local.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v2_no_opts/ui5-mock.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v2_no_opts/ui5.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v4_cap/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/alp_v4_cap_typescript/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/feop_v4_cap/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v2_no_url/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v2_no_url/ui5-local.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v2_no_url/ui5-mock.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v2_no_url/ui5.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v4_no_ui5_version/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v4_no_ui5_version/ui5-local.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v4_no_ui5_version/ui5-mock.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v4_no_ui5_version/ui5.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v4_no_ui5_version/webapp/manifest.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/headless/lrop_v4_no_ui5_version/webapp/test/integration/pages/JourneyRunner.js
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v2/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v2/ui5-local.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v2/ui5-mock.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v2/ui5.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v4/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v4/ui5-local.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v4/ui5-mock.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v4/ui5.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v4/webapp/test/integration/pages/JourneyRunner.js
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v4_cap/app/lrop_v4_cap/ui5.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v4_cap/app/lrop_v4_cap/webapp/test/integration/pages/JourneyRunner.js
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v4_cap/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-elements/expected-output/lrop_v4_version_not_specified/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-freestyle/expected-output/headless/simpleflightnodatasource/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-freestyle/expected-output/headless/simpleflightnodatasource/ui5-local.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-freestyle/expected-output/headless/simpleflightnodatasource/ui5.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-freestyle/expected-output/headless/simpleflightv2/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-freestyle/expected-output/headless/simpleflightv2/ui5-local.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-freestyle/expected-output/headless/simpleflightv2/ui5-mock.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-freestyle/expected-output/headless/simpleflightv2/ui5.yaml
#	packages/fiori-app-sub-generator/test/int/fiori-freestyle/expected-output/simple_cap/package.json
#	packages/fiori-app-sub-generator/test/int/fiori-freestyle/expected-output/simple_v4/ui5-mock.yaml
#	packages/fiori-docs-embeddings/data/index.js
#	packages/fiori-docs-embeddings/tsconfig.json
#	packages/fiori-elements-writer/CHANGELOG.md
#	packages/fiori-elements-writer/package.json
#	packages/fiori-freestyle-writer/CHANGELOG.md
#	packages/fiori-freestyle-writer/package.json
#	packages/fiori-generator-shared/CHANGELOG.md
#	packages/fiori-generator-shared/package.json
#	packages/fiori-mcp-server/.claude-plugin/plugin.json
#	packages/fiori-mcp-server/CHANGELOG.md
#	packages/fiori-mcp-server/package.json
#	packages/fiori-mcp-server/scripts/bundle.mjs
#	packages/fiori-mcp-server/server.json
#	packages/fiori-mcp-server/test/unit/tools/functionalities/fetch-service-metadata/service-metadata.test.ts
#	packages/fiori-mcp-server/tsconfig.json
#	packages/flp-config-inquirer/CHANGELOG.md
#	packages/flp-config-inquirer/package.json
#	packages/flp-config-sub-generator/CHANGELOG.md
#	packages/flp-config-sub-generator/package.json
#	packages/generator-adp/CHANGELOG.md
#	packages/generator-adp/package.json
#	packages/generator-adp/tsconfig.json
#	packages/generator-odata-downloader/CHANGELOG.md
#	packages/generator-odata-downloader/jest.config.mjs
#	packages/generator-odata-downloader/package.json
#	packages/generator-odata-downloader/src/data-download/prompts/prompt-helpers.ts
#	packages/generator-odata-downloader/src/data-download/utils.ts
#	packages/generator-odata-downloader/src/telemetry/index.ts
#	packages/generator-odata-downloader/test/test-data/test-apps/purchaseorder/webapp/localService/mockdata/PPS_PurOrdItemHierarchy.js
#	packages/generator-odata-downloader/test/test-data/test-apps/purchaseorder/webapp/localService/mockdata/PPS_PurchaseOrderItem.js
#	packages/generator-odata-downloader/tsconfig.json
#	packages/inquirer-common/CHANGELOG.md
#	packages/inquirer-common/package.json
#	packages/launch-config/CHANGELOG.md
#	packages/launch-config/package.json
#	packages/mockserver-config-writer/CHANGELOG.md
#	packages/mockserver-config-writer/package.json
#	packages/nodejs-utils/CHANGELOG.md
#	packages/nodejs-utils/package.json
#	packages/odata-annotation-core/tsconfig.json
#	packages/odata-service-inquirer/CHANGELOG.md
#	packages/odata-service-inquirer/package.json
#	packages/odata-service-writer/CHANGELOG.md
#	packages/odata-service-writer/package.json
#	packages/preview-middleware-client/CHANGELOG.md
#	packages/preview-middleware-client/package.json
#	packages/preview-middleware-client/src/cpe/odata-health/odata-health-checker.ts
#	packages/preview-middleware-client/test/__mock__/sap/ui/rta/plugin/ExtendControllerPlugin.ts
#	packages/preview-middleware-client/test/unit/adp/add-fragment.test.ts
#	packages/preview-middleware-client/test/unit/adp/change-file-validator.test.ts
#	packages/preview-middleware-client/test/unit/adp/controllers/AddCustomFragment.controller.test.ts
#	packages/preview-middleware-client/test/unit/adp/controllers/AddTableColumnFragments.controller.test.ts
#	packages/preview-middleware-client/test/unit/adp/controllers/ControllerExtension.controller.test.ts
#	packages/preview-middleware-client/test/unit/adp/dialog-factory.test.ts
#	packages/preview-middleware-client/test/unit/adp/extend-controller.test.ts
#	packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v2.test.ts
#	packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v4.test.ts
#	packages/preview-middleware-client/test/unit/cpe/additional-change-info/add-xml-additional-info.test.ts
#	packages/preview-middleware-client/test/unit/cpe/context-menu-service.test.ts
#	packages/preview-middleware-client/test/unit/cpe/init.test.ts
#	packages/preview-middleware-client/test/unit/flp/enableFakeConnector.test.ts
#	packages/preview-middleware-client/test/unit/utils/core.test.ts
#	packages/preview-middleware-client/test/unit/utils/info-center-message.test.ts
#	packages/preview-middleware-client/tsconfig.json
#	packages/preview-middleware/CHANGELOG.md
#	packages/preview-middleware/package.json
#	packages/project-access/CHANGELOG.md
#	packages/project-access/package.json
#	packages/project-input-validator/CHANGELOG.md
#	packages/project-input-validator/package.json
#	packages/project-integrity/CHANGELOG.md
#	packages/project-integrity/package.json
#	packages/reload-middleware/CHANGELOG.md
#	packages/reload-middleware/package.json
#	packages/repo-app-import-sub-generator/CHANGELOG.md
#	packages/repo-app-import-sub-generator/package.json
#	packages/sap-systems-ext/tsconfig.json
#	packages/system-access/CHANGELOG.md
#	packages/system-access/package.json
#	packages/system-access/test/__mocks__/index.ts
#	packages/telemetry/CHANGELOG.md
#	packages/telemetry/package.json
#	packages/ui-components/CHANGELOG.md
#	packages/ui-components/package.json
#	packages/ui-components/src/components/UISection/_variables.scss
#	packages/ui-components/src/styles/_typography.scss
#	packages/ui-prompting/CHANGELOG.md
#	packages/ui-prompting/package.json
#	packages/ui-service-inquirer/CHANGELOG.md
#	packages/ui-service-inquirer/package.json
#	packages/ui-service-sub-generator/CHANGELOG.md
#	packages/ui-service-sub-generator/package.json
#	packages/ui5-application-inquirer/CHANGELOG.md
#	packages/ui5-application-inquirer/package.json
#	packages/ui5-application-writer/CHANGELOG.md
#	packages/ui5-application-writer/package.json
#	packages/ui5-config/CHANGELOG.md
#	packages/ui5-config/package.json
#	packages/ui5-config/scripts/download-ui5yaml-schema.js
#	packages/ui5-config/src/ui5config.ts
#	packages/ui5-library-inquirer/CHANGELOG.md
#	packages/ui5-library-inquirer/package.json
#	packages/ui5-library-reference-inquirer/CHANGELOG.md
#	packages/ui5-library-reference-inquirer/package.json
#	packages/ui5-library-reference-sub-generator/CHANGELOG.md
#	packages/ui5-library-reference-sub-generator/package.json
#	packages/ui5-library-reference-writer/CHANGELOG.md
#	packages/ui5-library-reference-writer/package.json
#	packages/ui5-library-sub-generator/CHANGELOG.md
#	packages/ui5-library-sub-generator/package.json
#	packages/ui5-library-writer/CHANGELOG.md
#	packages/ui5-library-writer/package.json
#	packages/ui5-proxy-middleware/CHANGELOG.md
#	packages/ui5-proxy-middleware/package.json
#	packages/ui5-test-writer/CHANGELOG.md
#	packages/ui5-test-writer/package.json
#	packages/ui5-test-writer/src/fiori-elements-opa-writer.ts
#	packages/ui5-test-writer/src/index.ts
#	packages/ui5-test-writer/src/utils/opaQUnitUtils.ts
#	packages/ui5-test-writer/test/integration/fiori-elements.test.ts
#	packages/ui5-test-writer/test/unit/__snapshots__/fiori-elements.test.ts.snap
#	packages/ui5-test-writer/test/unit/utils/opaQUnitUtils.test.ts
#	packages/ui5-test-writer/tsconfig.json
#	pnpm-lock.yaml
#	pnpm-workspace.yaml
#	tests/integration/adaptation-editor/CHANGELOG.md
#	tests/integration/adaptation-editor/package.json
#	tsconfig.json

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme

* origin/main: (180 commits)
  chore: apply latest changesets
  fix(app-config-writer): no card generator for CAP node without cds-plugin-ui5 (#4930)
  chore: apply latest changesets
  fix(eslint-plugin): deduplicate trivial labels in diagnostics for shared text properties (#4919)
  chore: apply latest changesets
  fix(ui5-test-writer): Object pages navigation now expect any number of OPs (#4918)
  chore: apply latest changesets
  fix(ui5-test-writer): handle ConnectedFields and FieldGroup wrappers (#4878)
  chore: apply latest changesets
  fix(ui5-test-writer): Incorrect arguments used in generated TS test code (#4916)
  feat(fiori-docs-embeddings): add sap.fe.test JSDoc API as embedded data source (#4917)
  chore: apply latest changesets
  Feat(eslint-plugin-fiori-tools): enable no-live-mode rule for OData V2 applications (#4915)
  chore: apply latest changesets
  fix(ui5-test-writer): Use const in JS templates (#4925)
  chore: apply latest changesets
  feat(fiori-mcp-server): skill for fiori opa5 development (#4896)
  chore: apply latest changesets
  fix: conditional usage of inbound change types (#4920)
  chore: apply latest changesets
  ...

devinea committed:
fix changesets

devinea committed:
test(ui-components): replace enzyme with react testing library

Migrate all remaining 32 enzyme-based test files in @sap-ux/ui-components
to @testing-library/react. All 614 tests pass.

devinea committed:
revert

devinea committed:
lint

devinea committed:
fix flakey test

devinea committed:
lint clean-up

devinea committed:
chore: revert unrelated workflow change

devinea committed:
review clean-up

devinea committed:
fixes

devinea committed:
chore(ui-components): remove enzyme packages from devDependencies

devinea committed:
test(ui-components): address Sonar S5906 - use toHaveLength, toBeNull, toBeUndefined

devinea committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme

* origin/main:
  Feat(eslint-plugin-fiori-tools): enable multiview linting - ODataV4 (#4910)

marcovieth committed:
Merge branch 'main' into replace_enzyme

marcovieth committed:
test: fix Sonar issues

marcovieth committed:
test: UIDropdown: remove "if"; adapt duplicate test case (pre-existing bug)

marcovieth committed:
test: update component names in test descriptions for consistency

marcovieth committed:
test: add TODO comments to improve clarity on testing implementation details

marcovieth committed:
test: add TODO to improve UILabel styles test for better output validation

marcovieth committed:
test: remove unnecessary unmount calls in UITreeDropdown tests

marcovieth committed:
test: add TODO comments to improve clarity on testing implementation details

marcovieth committed:
test: remove conditions in tests which should be always true

marcovieth committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme

marcovieth committed:
chore: remove enzyme dependency from lockfiles

marcovieth committed:
Merge remote-tracking branch 'origin/main' into replace_enzyme

github-actions[bot] committed:
Linting auto fix commit

marcovieth commented:
In addition to the list of validating tests: https://github.com/SAP/open-ux-tools/pull/3459#issuecomment-3729288410

Post-migration test cleanup (since July 24)

- Remove unnecessary unmount calls (UITreeDropdown) — @testing-library/react handles cleanup automatically via afterEach, explicit unmount is redundant (e93bdabb1)
- Remove always-true conditions (UISplitButton, UITable, UITreeDropdown) — checks like if (x) { expect(x)... } that could never fail, left over from Enzyme's looser querying style (4d7752214)
- Update test description names (UICheckbox, UIChoiceGroup, UILabel, UISection, UISections, UISplitter, UITextfield) — align describe/it labels with actual component names for consistency (a7977bda1)
- Fix UIDropdown duplicate test — removed an if guard and adapted a duplicate test case (pre-existing bug in the migrated test) (841a2d442)
- Add TODO comments (UICallout, UIContextualMenu, UIDialog, UILink, UIMessageBar, UISmallButton, UITooltip, UIChoiceGroup, UIDefaultButton, UITextfield, UIToggle, UILabel) — mark places where tests only check element existence or class names rather than actual rendered output, flagging them for future improvement (5e3ab4c50, 6ddf9ea3c, d07430a68)
- Remove enzyme>cheerio override from pnpm-workspace.yaml — the last Enzyme dependency reference in the repo config (6841cfecc)

marcovieth commented:
Possible further improvements:
Render without props → immediate rerender with props (leftover from setProps)

During the Enzyme → react-testing-library migration, wrapper.setProps({ ... }) was mechanically translated to render(<Component />) followed immediately by rerender(<Component prop={...} />) with no assertions between them. Since no prop-update behavior is actually being tested, the initial render is redundant and can be collapsed into a single render(<Component prop={...} />).

| File	                | Affected tests|
| ----------- | ----------- |
| UICombobox.test.tsx	| ~25 cases (most tests in the file)| 
| UIDropdown.test.tsx	| 13 tests| 
| UISection.test.tsx	| 5 tests (height, className, title, collapsible, onScroll)| 
| UISections.test.tsx	| 5 tests (vertical, animation, height, splitterLayoutType, splitter beforeEach)| 
| UIMessageBar.test.tsx	| 4 tests (error, success, info, warning)| 
| UIToggle.test.tsx	| 4 tests (validation message: error-standard, error-inline, warning, info)| 
| UILink.test.tsx	| 3 tests (secondary, primary no underline, disabled)| 
| UITreeDropdown.test.tsx| 2 beforeEach blocks (value change, submenu offset)| 
| UITooltip.test.tsx	| 1 test (maxWidth custom)| 
| UIIcon.test.tsx	        | 1 test (classname)| 
| UICheckbox.test.tsx	| 1 test (error validation message)| 
| UITextfield.test.tsx	| 1 test (error custom component)| 
| UISearchBox.test.tsx   | 1 test| 
| UISplitter.test.tsx	| 1 test (vertical splitter)

--

In UITreeDropdown.test.tsx the rerender is intentional and correct.
For the other tests, render and re-render are combined with 44b49f9e449002ad1f5a21ef4ed4d6673d927378

marcovieth committed:
Merge branch 'main' into replace_enzyme

marcovieth committed:
Refactor tests to remove unnecessary rerenders

- Simplified test cases by directly rendering components with updated props instead of using rerender.

marcovieth committed:
Refactor unit tests for UILabel, UITextInput, and UIToggle components to improve style assertions

- Updated UILabel tests to directly assert rendered styles for disabled and required states.
- Simplified UITextInput tests by removing direct style function calls and replacing them with DOM assertions.
- Enhanced UIToggle tests to verify styles through rendered output rather than internal implementation details.

marcovieth committed:
Refactor unit tests for various components to improve style assertions and remove unnecessary mocks

marcovieth committed:
chore: add changeset for rebuilding bundle with updated preview-middleware-client

marcovieth committed:
Fix lint issues

marcovieth committed:
Refactor UISection tests to improve clarity and remove unnecessary assertions

marcovieth committed:
Refactor unit tests for UILoader, UIPersona, and UISearchBox components to improve clarity and consistency

marcovieth committed:
Refactor UISections tests to correct component name and improve test descriptions

marcovieth committed:
Add NODE_OPTIONS to ui-components configuration for experimental VM modules

marcovieth committed:
Refactor tests to use jest fake timers and improve async handling in multiple components

marcovieth committed:
Refactor UIActionButton test to simplify selector usage and improve clarity

marcovieth committed:
Refactor delay function in UIFlexibleTable tests to remove time parameter and simplify async handling

marcovieth committed:
Refactor UITable tests to skip cell navigation test and enhance validation logic with mock function

marcovieth committed:
Refactor UI component tests to improve clarity and accuracy in descriptions

marcovieth commented:
## Test Suite Improvements — `@sap-ux/ui-components`

### 1. Replaced Fluent UI internal class selectors
Replaced `.ms-*` class assertions (Fluent UI internals) with stable alternatives wherever possible:
- `button`, `input`, `svg`, semantic ARIA roles
- Component-owned CSS classes (`.ts-*`, `.ui-*`, `.flexible-table-*`, etc.)

Affected: `UIPersona`, `UILoader`, `UISearchBox`, `UISection`, `UISections`, `UIToggle`, `UITextfield`, `UITooltip` (render test), `UIActionButton`, `UIContextualMenu` (render test), `UILabel`, `UIChoiceGroup`, `UIDropdown`

### 2. Replaced real `setTimeout` waits with fake timers
Eliminated real timer delays that slowed down the suite:

| File | Before | After |
|------|--------|-------|
| `UITreeDropdown` | 8 × 100–300ms real waits | `jest.useFakeTimers` + `runOnlyPendingTimers` |
| `UITooltip` | 2 × 1000ms real waits | `legacyFakeTimers` + `runAllTimers` |
| `UIDialog` | 1 × 500ms real wait | `legacyFakeTimers` + `runOnlyPendingTimers` |
| `UICombobox` | 3 real waits (0ms, 20ms) | `requestAnimationFrame` mock + `advanceTimersByTime` |
| `UIFlexibleTable` | 5 × 200ms real waits | 0ms yield inside `act()` |
| `UITextfield` | 1 × 100ms real wait | `legacyFakeTimers` + `runAllTimers` |
| `CalloutCollisionTransform` | 3 × 1ms real waits | `legacyFakeTimers` + `runOnlyPendingTimers` |
| `UITable-helper` (`sleep`) | 110ms real wait | `jest.useFakeTimers` + `advanceTimersByTime` |

### 3. Fixed wrong/misleading test names
- `'Should render a Shell component'` → `'Should render a UISections/UISection component'`
- `'Test "collapsible" property'` → `'Test "layout" property'`
- `'Test spliter resize/toggle'` → `'Test splitter resize/toggle'` (typo)
- `'Test "splitter" resize'` (duplicate) → `'Test "splitter" resize vertical'`
- `'Should render a PropertiesPanel component'` → `'Should render a UISplitter component'`
- `'Should render a UISplitButton component - ...'` → descriptive names per behavior
- `'Scrollable - 50 width'` (duplicate) → `'Not scrollable - no offset'`
- `'getUIcontextualMenuCalloutStyles - pass maxWidth'` (duplicate) → `'getUIcontextualMenuCalloutStyles - pass maxWidth and custom styles'`

### 4. Removed redundant assertions
- Removed `querySelectorAll(...).toHaveLength(1)` when `querySelector(...).not.toBeNull()` already covers it
- Removed `afterEach(jest.clearAllMocks)` where no shared mocks existed

### 5. Fixed fake parametrization
- `getCalloutCollisionTransformationProps` — 4 identical test cases replaced with 4 proper branch-covering tests (`true/true`, `false/true`, `true/false`, `undefined/undefined`)
- `useDropdownAsMenuMinWidth` — removed duplicate `Default`/`False` test case

### 6. Improved test isolation and readability
- `UISplitButton`: renamed 4 identical test names to describe actual behavior
- `MessageWrapper`: replaced `JSON.parse(JSON.stringify(...))` clone with `{ ...obj, type: undefined }`
- `UICombobox` (`requestAnimationFrame`): replaced `async/await` + real timer with synchronous spy approach
- `UITreeDropdown`: removed unnecessary `async/await` from tests that no longer needed them

### 7. Improved dead/skipped tests
- `UITable` `'Cell navigation in edit mode'` — had no assertions and a `// must investigate` comment; changed to `it.skip` so it's visible as a known gap
- `UITable` `'Validate and focus'` — was checking `.ms-TextField` exists (always true); now verifies `validate` was called and `errorMessage` is set in component state

### 8. Source bug fix (maybe separate PR, not yet included)
- `addRowNumbers` in `UITable-helper.tsx` — crashes on empty array; fixed with `columns.length > 0` guard; test updated to assert correct behavior instead of documenting the crash

ManuelPastorini commented on packages/ui-components/test/unit/components/UIFlexibleTable.test.tsx:
```diff
@@ -77,141 +75,141 @@ describe('<UIFlexibleTable />', () => {
-            expect(titleCells.get(0).key).toBe(`title-cell-unknown`);
-            expect(titleCells.get(0).props.children).toBe(col.title);
-            expect(titleCells.get(0).props['id']).toBe(expectedId);
+            // isSpan=true renders key="title-cell-unknown"
```
commented code should re removed if not needed.
But: Seems we get a bit less specific by missing one assertion?

marcovieth replied:
React key props are internal to React — they're never rendered into the DOM, so there's no equivalent RTL assertion. The comment is documenting dropped coverage that can't be restored.

The two assertions on lines 190-191 (id and textContent) already cover the meaningful DOM behavior of that cell. Comment removed.

ManuelPastorini replied:
agreed

ManuelPastorini commented on packages/ui-components/test/unit/components/UIFlexibleTable.test.tsx:
```diff
@@ -227,173 +225,142 @@ describe('<UIFlexibleTable />', () => {
-            `);
+            setProps({ maxWidth: 1000 });
+            const root = container.querySelector(selectors.tableRoot) as HTMLElement;
+            expect(root).toBeTruthy();
```
Test is less specific than before - use toHaveLength(1)

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UIFlexibleTable.test.tsx:
```diff
@@ -793,114 +707,116 @@ describe('<UIFlexibleTable />', () => {
                 );
-                expect(action.length).toBe(1);
-                expect(action.get(0).props.children).toBe((rowIndex + 1).toString());
+                expect(action).toBeTruthy();
```
less specific than old test

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UICallout.test.tsx:
```diff
@@ -13,32 +20,22 @@ const { getNextElement: mockGetNextElement, getPreviousElement: mockGetPreviousE
-    it('Should render a UITooltip component', () => {
-        expect(wrapper.find('.ms-Callout').length).toEqual(1);
-        const style = getCalloutStyles();
+        expect(document.querySelector('.ms-Callout')).toBeTruthy();
```
expect is less specific than before

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -1,163 +1,220 @@
-            const buttons = wrapper.find(UIDefaultButton);
-            expect(wrapper.find(DialogFooter).length).toEqual(1);
+            const buttons = document.body.querySelectorAll('button');
+            expect(capturedFooterProps).toBeDefined();
```
new expect is less specific than original one

marcovieth replied:
DialogFooter is a React component, so Enzyme's .find(DialogFooter) traverses the React tree, but RTL has no equivalent. The mock capture (capturedFooterProps) is exactly the right RTL-era replacement here: the mock intercepts the render and records that DialogFooter was called with props.
toBeDefined() confirms that DialogFooter rendered. The DOM buttons.length check already validates the structural result.
Same comment for line 142, 150, 158, 165, 178.

ManuelPastorini replied:
To my understanding, using capturedFooterProps instead of DialogFooter is fine. However, the old test was checking for exactly one footer (by checking "length).toEqual(1)", while now we are only checking for toBeDefined, which would also pass in case of multiple footers! In this case we would not know, whether we have captured the properties from the right DialogFooter (it just stores the props, so the last one wins) => we should either check, that it is called exactly once, or confirm the props it stored!

Same comment for line 142, 150, 158, 165, 178 ;-)

marcovieth replied:
Good point. Done with 672f852
(Also for lines 142, 150, 158, 165, 178)

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -1,163 +1,220 @@
+            capturedFooterProps = undefined;
+            rerenderWith({ onCancel: undefined });
+            const buttons = document.body.querySelectorAll('button');
+            expect(capturedFooterProps).toBeDefined();
```
new expect is less specific than original one

marcovieth replied:
Check comment for line 134: The mock capture (capturedFooterProps) is used for RTL.

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -1,163 +1,220 @@
+            capturedFooterProps = undefined;
+            rerenderWith({ onAccept: undefined });
+            const buttons = document.body.querySelectorAll('button');
+            expect(capturedFooterProps).toBeDefined();
```
new expect is less specific than original one

marcovieth replied:
Check comment for line 134: The mock capture (capturedFooterProps) is used for RTL.

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -1,163 +1,220 @@
+            capturedFooterProps = undefined;
+            rerenderWith({ onAccept: undefined, onCancel: undefined });
+            const buttons = document.body.querySelectorAll('button');
+            expect(capturedFooterProps).toBeUndefined();
```
new expect is less specific than original one

marcovieth replied:
Check comment for line 134: The mock capture (capturedFooterProps) is used for RTL.

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -1,163 +1,220 @@
-            expect(wrapper.find('.dummyFooter').length).toEqual(1);
+            capturedFooterProps = undefined;
+            rerenderWith({ footer: <div className="dummyFooter"></div> });
+            expect(capturedFooterProps).toBeDefined();
```
new expect is less specific than original one

marcovieth replied:
Check comment for line 134: The mock capture (capturedFooterProps) is used for RTL.

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -1,163 +1,220 @@
             });
-            expect(wrapper.find(DialogFooter).length).toEqual(1);
-            expect(wrapper.find('UIDefaultButton.dummyButton').length).toEqual(3);
+            expect(capturedFooterProps).toBeDefined();
```
new expect is less specific than original one

marcovieth replied:
Check comment for line 134: The mock capture (capturedFooterProps) is used for RTL.

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -430,35 +473,33 @@ describe('<UIDialog />', () => {
-            expect(titleElement.length).toEqual(1);
+            rerenderWith({ modalProps });
+            const mainEl = document.body.querySelector(dialogSelectors.main);
+            expect(mainEl).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -430,35 +473,33 @@ describe('<UIDialog />', () => {
+            const mainEl = document.body.querySelector(dialogSelectors.main);
+            expect(mainEl).toBeTruthy();
+            const titleElement = document.body.querySelector(dialogSelectors.title);
+            expect(titleElement).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -430,35 +473,33 @@ describe('<UIDialog />', () => {
-            titleElement.simulate('mousedown');
-            expect(titleElement.length).toEqual(1);
+            fireEvent.mouseDown(titleElement!);
+            expect(document.body.querySelector(dialogSelectors.title)).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -430,35 +473,33 @@ describe('<UIDialog />', () => {
-            const titleElement = wrapper.find(dialogSelectors.title);
-            expect(titleElement.length).toEqual(1);
+            const mainEl = document.body.querySelector(dialogSelectors.main);
+            expect(mainEl).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -430,35 +473,33 @@ describe('<UIDialog />', () => {
+            const mainEl = document.body.querySelector(dialogSelectors.main);
+            expect(mainEl).toBeTruthy();
+            const titleElement = document.body.querySelector(dialogSelectors.title);
+            expect(titleElement).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UIDialog.test.tsx:
```diff
@@ -430,35 +473,33 @@ describe('<UIDialog />', () => {
-            titleElement.simulate('mousedown');
-            expect(titleElement.length).toEqual(1);
+            fireEvent.mouseDown(titleElement!);
+            expect(document.body.querySelector(dialogSelectors.title)).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UIFlexibleTable.test.tsx:
```diff
@@ -793,114 +707,116 @@ describe('<UIFlexibleTable />', () => {
-                        .map((i) => i.props.actionName);
-                    expect(actions).toEqual(['up', 'down']);
+                    const rowActions = row.querySelector(selectors.rowHeaderActionsContainer);
+                    expect(rowActions).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini replied:
Not solved - you just replaced "toBeTruthy" with "not.toBeNull" here. The old test was not only checking, that the actions exist, but actually that their names were 'up' and 'down' - this not tested now.

marcovieth replied:
Done with commit 65ecd62

ManuelPastorini commented on packages/ui-components/test/unit/components/UIFlexibleTable.test.tsx:
```diff
@@ -793,114 +707,116 @@ describe('<UIFlexibleTable />', () => {
-                const disabled = action.getElement().props.disabled;
-                expect(!!disabled).toEqual(rowIndex === 1);
+                ) as HTMLButtonElement | null;
+                expect(action).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UIFlexibleTable.test.tsx:
```diff
@@ -962,86 +864,96 @@ describe('<UIFlexibleTable />', () => {
-            expect(noData.length).toEqual(1);
+            setProps({ noDataText: <div className="customNoData"></div> });
+            const noData = container.querySelector('.customNoData');
+            expect(noData).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UILoader.test.tsx:
```diff
@@ -1,50 +1,28 @@
-        expect(wrapper.find('.ms-Spinner-circle').length).toEqual(1);
-        expect(wrapper.find(Overlay).length).toEqual(0);
+        const { container } = renderLoader();
+        expect(container.firstChild).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented on packages/ui-components/test/unit/components/UISplitter.test.tsx:
```diff
@@ -29,19 +26,17 @@ describe('<Splitter />', () => {
-        expect(wrapper.find('.splitter--vertical').length).toEqual(0);
+    it('Should render a UISplitter component', () => {
+        const { container } = render(<UISplitter type={UISplitterType.Resize} onResize={onResize} />);
+        expect(container.querySelector('.splitter')).toBeTruthy();
```
new expect is less specific than original one

marcovieth replied:
Done with commit 0bf442f

ManuelPastorini commented:
At some places, the migrated assertions are less specific than the original ones. These should be correct before merging to not loose the information.
Apart from that, the PR looks good to me.

ManuelPastorini commented on packages/ui-components/test/unit/components/UICombobox.test.tsx:
```diff
@@ -165,239 +183,246 @@ describe('<UIComboBox />', () => {
-            await new Promise((resolve) => setTimeout(resolve));
-            expect(requestAnimationFrameSpy).toHaveBeenCalledTimes(2);
-            expect((input.getDOMNode() as HTMLInputElement).selectionEnd).toBe(2);
+        it('Test onInput value selection', () => {
```
Why the toHaveBeenCalledTimes(2) assertion was dropped

The old test used an unmocked requestAnimationFrame spy and counted calls cumulatively across both blocks (block 1 → 1 call, block 2 → 2 calls total). The new test replaces this with a per-block relative count (toHaveBeenCalledTimes(1) after the second fireEvent.input), which is equivalent.

Why the expect(selections).toBe(4) assertion was dropped

In the old Enzyme test, selectionEnd === 4 after block 1 was an artefact of Enzyme's simulate + await setTimeout combination: jsdom happened to execute the rAF callback automatically in that environment. This assertion was testing a side-effect of the wrong scenario — the cursor was actually at position 0 (jsdom default), not 2 as intended, so the selectionEnd !== value.length guard in setCaretPosition fired for the wrong reason. The new test replaces this entire scenario with an explicit "cursor at end" case (selectionEnd = 4 on 'test') and asserts that no rAF is scheduled at all (toHaveBeenCalledTimes(0)), which directly matches the if (selectionEnd !== input.value.length) guard in the implementation.
